### ISSUE TITLE
Complete from specified sources, hopefully by remapping `compl-tag`, `compl-dictionary`, etc.

### ISSUE THREAD

**Issue by yjyao** [neoclide/coc-sources#61]
## overview

vim has built-in functionality to complete from *specified sources only*:
- tags (with `i_CTRL-X_CTRL-]` aka `compl-tag`)
- dictionary (with `i_CTRL-X_CTRL-K` aka `compl-dictionary`)
- thesaurus (with `i_CTRL-X_CTRL-T`)
- etc.

how do we remap, e.g., `compl-tag` to use `coc-tag`'s completion?

## steps to reproduce

1. install `coc.nvim` and `coc-tag`.
2. open a file where `:echo tagfiles()` is non-empty.
3. create a tag `my-fancy-tag` and refresh the tags file.
4. write in the file `manifest`.
5. type `mft` should show completion options `my-fancy-tag  [T]` and `manifest  [A]`.
6. type `mft` then press `<C-x><C-]>` (<kbd>Ctrl</kbd>+<kbd>x</kbd> and then <kbd>Ctrl</kbd>+<kbd>]</kbd>)

## expected results

with `<C-x><C-]>`, `mft` should complete to `my-fancy-tag  [T]` but not `manifest  [A]`.

## actual results

after pressing `<C-x><C-]>`, the completion list reduces to tags that start with `mft` only.

similarly, if you enter `m` then press `<C-x><C-]>`, `my-fancy-tag  [T]` will show up; however, if you press `f`, the `compl-tag` function won't find a match and will fallback to coc's completion which will also include `manifest  [A]`.

 basically, `<C-x><C-]>` triggers the built-in `compl-tag` function and does not use `coc-tag`'s tag completion.

**Comment by chemzqm** [comment]
`:h coc#start()`
